systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -1450,6 +1457,157 @@ void testClientsCACertRenew() {
+        final String kafkaClientsPodName =
+                ResourceManager.kubeClient().listPodsByPrefixInName(clusterName + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+                .withUsingPodName(kafkaClientsPodName)
+                .withTopicName(TOPIC_NAME)
+                .withNamespaceName(NAMESPACE)
+                .withClusterName(clusterName)
+                .withKafkaUsername(USER_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                .build();
+
+        LOGGER.info("Check for certificates used within kafka pod internal clients (producer/consumer)");
+        List<VolumeMount> volumeMounts = ResourceManager.kubeClient().listPodsByPrefixInName(clusterName + "-" + Constants.KAFKA_CLIENTS).get(0).getSpec().getContainers().get(0).getVolumeMounts();
+        for (VolumeMount vm : volumeMounts) {
+            if (vm.getMountPath().contains("user-secret-" + internalKafkaClient.getKafkaUsername())) {
+                assertThat("UserCert Issuer DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "issuer", STRIMZI_INTERMEDIATE_CA));
+                assertThat("UserCert Subject DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "subject", STRIMZI_TEST_CLIENTS_CA));
+
+            } else if (vm.getMountPath().contains("cluster-ca-" + internalKafkaClient.getKafkaUsername())) {
+                assertThat("ClusterCA Issuer DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "issuer", STRIMZI_INTERMEDIATE_CA));
+                assertThat("ClusterCA Subject DN in clients pod is incorrect!", checkMountVolumeSecret(kafkaClientsPodName,
+                        vm, "subject", STRIMZI_TEST_CLUSTER_CA));
+            }
+        }
+
+        LOGGER.info("Checking produced and consumed messages via TLS to pod:{}", kafkaClientsPodName);
+        internalKafkaClient.checkProducedAndConsumedMessages(
+                internalKafkaClient.sendMessagesTls(),
+                internalKafkaClient.receiveMessagesTls()
+        );
+    }
+
+    boolean checkMountVolumeSecret(String podName, VolumeMount volumeMount, String principalDNType, String expectedPrincipal) {
+        String dn = ResourceManager.cmdKubeClient().execInPod(podName, "/bin/bash", "-c",
+                "openssl x509 -in " + volumeMount.getMountPath() + "/ca.crt -noout -" + principalDNType).out().strip();
+        String certOutIssuer = dn.substring(principalDNType.length() + 3).replace("/", ",");
+        return SystemTestCertManager.containsAllDN(certOutIssuer, expectedPrincipal);
+    }
```

Comment:
If you decide to delete the volumes check, remove also this please.

Reply:
yup, I am 50/50 on deletion of this check... as it's already done - everyone says the same.
I think, that for this PR I will keep it there, and for new one - validity of certs - If I will be able to reuse it, I would, else I will remove it.

Reply:
Copy paste it to another PR please. When you'll don't use it there, delete it. From MPOV it's not quite good to have some unused methods in code.